tests/testthat/test-combine.R

```diff
@@ -0,0 +1,19 @@
+context("combine")
+
+test_that("combination keeps class and order", {
+  expect_identical(c(hms(1), hms(2)), hms(1:2))
+})
+
+test_that("combination coerces to hms", {
+  expect_identical(c(hms(1), 2), hms(1:2))
+  expect_identical(c(hms(1), "00:00:02"), hms(1:2))
```

Comment:
I'd expect this to throw an error. Base R converts to `NA` and warns, but we can be stricter here (and relax later if necessary):

``` r
class(as.POSIXct("2017-12-12 12:34:56 CEST"))
#> [1] "POSIXct" "POSIXt"
c(Sys.time(), "2017-12-12 12:34:56 CEST")
#> Warning in as.POSIXlt.POSIXct(x, tz): NAs introduced by coercion
#> [1] "2017-08-03 09:32:19 CEST" NA
```

Reply:
The behaviour for `difftime` is different though:

```R
c(as.difftime("20:00:00", units="sec"), "00:00:10")
#> Time differences in secs
#> [1] 72000    10
```
I think we want to use `as.hms()` inside `c()`, but `as.hms()` will converts strings as well.
If we coerce to numeric, e.g.: 

```R
c.hms <- function(x, ...) {
  hms(c(as.numeric(x), as.numeric(...)))
}
```
Thoughts?
We risk having inconsistent behaviours:
```R
d <- as.difftime("20:00:00")
c(hms(1), d)
#> 00:00:01
#> 00:00:20
```

Reply:
Thanks. I wasn't aware of the difftime behavior.